### PR TITLE
Merge AI and Master branches

### PR DESCRIPTION
**Pull request by nzlz**
The goal of this PR is that everyone at Acutronic Robotics ends up using the same branch, `master`. 

The main differences with AI branch:
- **We still use cognition**. Since we are the only ones using it, we should be able to push our changes here. We are using the correct topic naming convention. (The reason behing still depending on cognition is that rclpy multiple callbacks are turning to be more inconsistent in terms of speed: [ROS Answers](https://answers.ros.org/question/320474/ros2-correct-usage-of-spin_once-to-receive-multiple-callbacks-synchronized).
- **We need custom URDFs**. The training plugin is different form the real one, so must be called differently from the URDF. We also need a visual blue point in the middle of the gripper to visually observe the accuracy of the learnt policy. @YueErro Can you take care of creating xacro files for the new urdfs inside /urdf/reinforcement_learning? Could this be dependent on the base mara+140 gripper xacro file somehow?
- **We use a different plugin for training**. This plugin is optimized for fast training, still lot of room for improvement. But important thing here is we do not change anything, only add a new plugin to the list.
- **New spawner script with input arg**. _mara_utils_scripts/scripts/spawn_mara_arg.py_ takes the urdf path as input argument, which avoids having the need to create a new spawn-script for each urdf. The usage in a ROS2 LaunchDescription object is the following:
     ```python3
     Node(package='mara_utils_scripts', node_executable='spawn_mara_arg.py', arguments=["reinforcement_learning/mara_robot_gripper_140_train.urdf"], output='screen')```

- **Removed warnings from the mara_gazebo_joint_plugin**.
- **Changed gazebo_ros_pkgs branch**. Update the branch to _acutronicrobotics/gazebo_ros_pkgs (ros2_gym-gazebo2)_, which is the same as `crystal` but with the [properties plugin](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/868) included.

---
### Concerns

The run plugin (mara_gazebo_joint_plugin) used to be smoother in the end of the execution. This could be due to the fact that we are trying a policy learnt in different urdfs, needs testing.

Please share your thoughts.

### CODE REVIEW

**Comment by nzlz** [comment]
Update. The run driver is doing fine, the bad policy execution was caused by a bug in the environment. Not related to this PR.

File: hros_cognition_mara_components/config/link_order_real.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+motors:
+  - hros_actuation_servomotor_000A3500010E:axis2
```
[the remark this answers is not in the document]
Comment (nzlz) [reply hros_cognition_mara_components/config/link_order_real.yaml]: This is using the same topics as the real robot. I guess the read Mara driver topics should be renamed too.

File: hros_cognition_mara_components/config/module_order.yaml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 motors:
-  - erleAlexATX2
-  - erleAlexATX
+  - hros_actuation_servomotor_0000000000012
```
[the remark this answers is not in the document]
Comment (nzlz) [reply hros_cognition_mara_components/config/module_order.yaml]: No idea what is `module_order.yaml` used for. We can just rename it I guess.

File: mara_gazebo_plugins/include/mara_gazebo_plugins/mara_gazebo_joint_plugin.hpp [hunk 3]
```diff
@@ -94,7 +94,7 @@ namespace gazebo_plugins
     void timer_motor_state_msgs();
     std::shared_ptr<rclcpp::TimerBase> timer_motor_state_;
 
-    void readfullFile(std::string file_to_read, hrim_generic_srvs::srv::Simulation3D& msg_sim_3d);
+    //void readfullFile(std::string file_to_read, hrim_generic_srvs::srv::Simulation3D& msg_sim_3d);
```
[the remark this answers is not in the document]
Comment (nzlz) [reply mara_gazebo_plugins/include/mara_gazebo_plugins/mara_gazebo_joint_plugin.hpp]: Ill remove all the unused code.

File: hros_cognition_mara_components/config/link_order_real.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rkojcev) [reply hros_cognition_mara_components/config/link_order_real.yaml]: so what are we using in the real should also match here as well, especially in the link_order_real

File: hros_cognition_mara_components/config/module_order.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (LanderU) [reply hros_cognition_mara_components/config/module_order.yaml]: Yes, use hrim_...

File: hros_cognition_mara_components/config/link_order_real.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LanderU) [reply hros_cognition_mara_components/config/link_order_real.yaml]: The new release uses hrim_...

File: hros_cognition_mara_components/config/module_order.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rkojcev) [reply hros_cognition_mara_components/config/module_order.yaml]: its only used here: https://github.com/AcutronicRobotics/MARA/blob/ai_v2/mara_gazebo/launch/module.launch.py#L48-L49

[the remark this answers is not in the document]
Comment (nzlz) [reply hros_cognition_mara_components/config/module_order.yaml]: Renamed.

File: hros_cognition_mara_components/config/link_order_real.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LanderU) [reply hros_cognition_mara_components/config/link_order_real.yaml]: Change this please. And LGTM!

[the remark this answers is not in the document]
Comment (rkojcev) [reply hros_cognition_mara_components/config/link_order_real.yaml]: renamed

**Comment by LanderU** [comment]
LGTM!

**Pull request merged**